data-transforms/flatten/transform.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/bcicen/jstream"
+	"github.com/redpanda-data/redpanda/src/transform-sdk/go"
+	"io"
+	"os"
+)
+
+const DELIM_KEY = "RP_FLATTEN_DELIM"
+const DELIM_DEFAULT = "."
+
+func main() {
+	delim, present := os.LookupEnv(DELIM_KEY)
+	if !present {
+		delim = DELIM_DEFAULT
+	}
+	fmt.Println("Using delimitter: ", delim)
+
+	buffer := bytes.NewBuffer(make([]byte, 1024))
+	fn := func(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+		return doFlatten(e, buffer, delim)
+	}
+
+	redpanda.OnRecordWritten(fn)
+	fmt.Println("Registered flatten function")
```

Comment:
In the latest version of the broker, this should never be printed. The above call is a blocking one.

Reply:
Do we have a non-blocking log api exposed?

Reply:
what are you asking for? The problem is that OnRecordWritten blocks forever, logging should work fine

Reply:
Oh that wasn't clear. I thought your comment was saying `fmt.Println` is a blocking call.

Reply:
Ah sorry 'bout that!